api/views/workflow.py

```diff
@@ -49,12 +49,16 @@ class SourceViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = SourceSerializer
     permission_classes = [DjangoModelPermissionsOrAnonReadOnly, ]
 
-    queryset = Source.objects.none()  # Required for DjangoModelPermissions
+    queryset = Source.objects.exclude(icon='').exclude(is_deleted=True)  # Required for DjangoModelPermissions
 
     VALID_IMAGE_TYPES = ('image/png', 'image/jpeg')
 
     def get_queryset(self):
-        return Source.objects.exclude(icon='').exclude(is_deleted=True)
+        long_title = self.request.query_params.get('long_title', None)
+
+        if long_title is None:
+            return self.queryset
+        return self.queryset.filter(long_title=long_title)
```

Comment:
Since it only checks field equality, couldn't we use `DjangoFilterBackend` instead of a custom `get_queryset`?